src/parser.rs

```diff
@@ -12,6 +12,7 @@ use crate::{
     flakeref::{FlakeRef, FlakeRefType, LocationParamKeys, LocationParameters, TransportLayer},
 };
 
+// TODO: use a param-specific parser, handle the inversion specificially
```

Comment:
This came up becasue one of the tests, which checks the parameter parsing, doesn't have matching behavior with left-to-right parsing of the `nom` approach